app/frontend/src/features/profile/actions/PendingRequest.tsx

```diff
@@ -0,0 +1,61 @@
+/* has to check whether request is sent or received
+if sent, return a simple disabled "pending request" button
+if recieved, return the response button*/
+import { SetMutationError } from "features/connections/friends";
+import useFriendRequests from "features/connections/friends/useFriendRequests";
+import React from "react";
+
+import PendingFriendRequestSent from "./PendingFriendRequestSent";
+import RespondToFriendRequestProfileButton from "./RespondToFriendRequestProfileButton";
+
+interface PendingRequestProps {
+  userId: number;
+  setMutationError: SetMutationError;
+}
+
+function CheckSentRequests(userId: number) {
+  const friendRequestsSent = useFriendRequests("sent").data;
+  const friendRequestToUser = friendRequestsSent?.find(
+    (e) => e.userId === userId
+  );
+  if (friendRequestToUser !== undefined) {
+    const request = friendRequestToUser;
+    return request;
+  } else {
+    return false;
+  }
```

Comment:
```suggestion
  return friendRequestToUser ? friendRequestToUser : false;
```
You could argue that what you wrote is more clear, and it's better for code to be clear and easily understandable rather than just have fewer lines. However in js/ts we abuse truthiness/falsiness pretty often so I would say the suggestion is usual enough to make it more concise like this.

Reply:
We could even simplify this further by doing `!!friendRequestToUser`? That way, it's returning the same data type (as a boolean) which makes the intent clearer that we're checking if a friend request has been sent or not, rather than interested in the full object itself in this context.

I think this logic is only needed for checking request received based on my other comment above.

Reply:
I changed it to the suggestion by @lucaslcode because the purpose is not just to check whether a request has been sent or received, but also return the friend request object so its `friendRequestId` and `state` properties can be accessed.

Reply:
Ha also ignore what I said here then, this may soon be irrelevant with the updated API!